lisp/lib.lisp

```diff
@@ -256,3 +256,89 @@ Example:
          (reducer (fn (fst nxt)
                      `((fn (_) (when _ ,fst)) ,nxt))))
         (iterator::reduce reducer first-rev rest-rev)))
+
+(defn mkli
```

Comment:
Does this make more sense in shell.lisp?  Not a big deal either way so leave if you want to.